--- openapi/build/greetings/build.gradle
+++ openapi/build/greetings/build.gradle
@@ -18,6 +18,7 @@
 }

 apply plugin: 'kotlin'
+apply plugin: 'maven-publish'

 repositories {
     mavenCentral()
@@ -34,3 +35,17 @@
     compile "com.squareup.okhttp3:okhttp:4.2.2"
     testCompile "io.kotlintest:kotlintest-runner-junit5:3.1.0"
 }
+
+task sourcesJar(type: Jar, dependsOn: classes) {
+    classifier = 'sources'
+    from sourceSets.main.allSource
+}
+
+publishing {
+    publications {
+        mavenJava(MavenPublication) {
+            from components.java
+            artifact sourcesJar
+        }
+    }
+}
